enterprise-suite/gotests/util/prometheus/prometheus.go

```diff
@@ -32,8 +33,12 @@ type Connection struct {
 
 func (p *Connection) Query(query string) (*PromResponse, error) {
 	addr := fmt.Sprintf("%v/api/v1/query?query=%v", p.url, url.QueryEscape(query))
-
-	resp, err := http.Get(addr)
+	timeout := time.Duration(45 * time.Second)
+	client := http.Client{
+		Timeout: timeout,
+	}
+	// client.Get(url)
+	resp, err := client.Get(addr)
```

Comment:
I think it's good to have a timeout here (I believe the default is no timeout). But this won't affect the flake you saw.